Sources/ProjectSpec/TargetSource.swift

```diff
@@ -100,12 +142,42 @@ extension TargetSource: JSONObjectConvertible {
         excludes = jsonDictionary.json(atKeyPath: "excludes") ?? []
         type = jsonDictionary.json(atKeyPath: "type")
         optional = jsonDictionary.json(atKeyPath: "optional") ?? false
+        
         if let string: String = jsonDictionary.json(atKeyPath: "buildPhase") {
-            if let buildPhase = BuildPhase(rawValue: string) {
-                self.buildPhase = buildPhase
-            } else {
-                throw SpecParsingError.unknownSourceBuildPhase(string)
-            }
+            buildPhase = try BuildPhase(string: string)
+        } else if let dict: JSONDictionary = jsonDictionary.json(atKeyPath: "buildPhase") {
+            buildPhase = try BuildPhase(jsonDictionary: dict)
+        }
+    }
+}
+
+extension TargetSource.BuildPhase {
+    
+    public init(string: String) throws {
+        switch string {
+        case "sources": self = .sources
+        case "headers": self = .headers
+        case "resources": self = .resources
+        case "copyFiles":
+            throw SpecParsingError.invalidSourceBuildPhase("copyFiles must specify directory")
```

Comment:
Should this be `copyFiles must specify a \"destination\" and optional \"subPath\"`

Reply:
Yes, that's more descriptive.